dev-packages/ovsx-client/src/ovsx-http-client.ts

```diff
@@ -43,10 +47,20 @@ export class OVSXHttpClient implements OVSXClient {
     }
 
     protected async requestJson<R>(url: string): Promise<R> {
-        return RequestContext.asJson<R>(await this.requestService.request({
-            url,
-            headers: { 'Accept': 'application/json' }
-        }));
+        for (let i = 0; i < 10; i++) {
```

Comment:
I'm not sure about the logic here: repeating the request when we're already running into the rate limit seems counterproductive. Wouldn't the right answer be to abort the task and for the client of this API to adjust the rate limit?

Reply:
Yes, I've removed the loop and added a warning/log for this.

Reply:
`RequestContext.asJson` already throws an error in that case. All non `2xx` codes will be thrown as errors in there.